source/custom/nixnetsocket_service.custom.cpp

```diff
@@ -1,2 +1,94 @@
+#include <nixnetsocket/nixnetsocket_service.h>
+
+#include <atomic>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+#include "xnetsocket_converters.h"
+
 namespace nixnetsocket_grpc {
+
+// Returns true if it's safe to use outputs of a method with the given status.
+inline bool status_ok(int32 status)
+{
+  return status >= 0;
+}
+
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+::grpc::Status NiXnetSocketService::GetSocketOption(::grpc::ServerContext* context, const GetSocketOptionRequest* request, GetSocketOptionResponse* response)
```

Comment:
This was getting to complicated to codegen in a straightforward way, so I opted to custom implement it. We might be able to codegen this instead but I'd rather get the service implementation reviewed first before attempting to get it generating.